components/org.wso2.carbon.identity.scim2.provider/src/main/java/org/wso2/carbon/identity/scim2/provider/util/SupportUtils.java

```diff
@@ -234,4 +237,22 @@ public static Response buildCreateUserResponse(SCIMResponse scimResponse) {
         }
         return buildResponse(scimResponse);
     }
+
+    /**
+     * This is used to set the flow and initiator in the identity context.
+     *
+     * @param flowName The name of the flow to set in the identity context.
+     */
+    public static void updateIdentityContextFlow(Flow.Name flowName) {
+
+        if (IdentityContext.getThreadLocalIdentityContext().isApplicationActor()) {
+            IdentityContext.getThreadLocalIdentityContext()
+                    .setFlow(new Flow.Builder().name(flowName).initiatingPersona(
+                            Flow.InitiatingPersona.APPLICATION).build());
+        } else {
```

Comment:
Let's check
`IdentityContext.getThreadLocalIdentityContext().isUserActor()` in here as well, since `IdentityContext.getThreadLocalIdentityContext().isApplicationActor()` is false when the actor is null in the context.